gitops/pac/setup.sh

```diff
@@ -0,0 +1,166 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+usage() {
+
+    printf "Usage: KUBECONFIG="path-to-kubeconfig" GITOPS_REPO="https://gitops.com/group/project" GIT_TOKEN="XXXXXXXXX" WEBHOOK_SECRET="YYYYYYYYYY" ./setup.sh\n\n"
+	
+    # Parameters
+    printf "The following parameters need to be passed to the script:\n"
+    printf "KUBECONFIG: the path to the kubeconfig file used to connect to the cluster where Pipelines as Code will be installed\n"
+    printf "GITOPS_REPO: the repository for which Pipelines as Code needs to be set up\n"
+    printf "GIT_TOKEN: personal access token to the git repository\n"
+    printf "WEBHOOK_SECRET: secret configured on the webhook used to validate the payload\n"
+    printf "OPERATOR_INSTALL (optional): the operator will be installed only if it is set to true. Pipelines as Code are installed together with the OpenShift Pipelines operator\n"
+    printf "PAC_HOST (optional): Hostname for the Pipelines as Code ingress if OPERATOR_INSTALL=true has been set\n\n"
+}
+
+check_params() {
+    KUBECONFIG="${KUBECONFIG:-}"
+    GITOPS_REPO="${GITOPS_REPO:-}"
+    GIT_TOKEN="${GIT_TOKEN:-}"
+    WEBHOOK_SECRET="${WEBHOOK_SECRET:-}"
+    if [[ -z "${KUBECONFIG}" ]]; then
+        printf "KUBECONFIG environment variable needs to be set\n\n"
+	usage
+	exit 1
+    fi
+    if [[ -z "${GITOPS_REPO}" ]]; then
+        printf "GITOPS_REPO environment variable needs to be set\n\n"
+	usage
+        exit 1
+    fi
+    if [[ -z "${GIT_TOKEN}" ]]; then
+        printf "GIT_TOKEN environment variable needs to be set\n\n"
+        usage
+        exit 1
+    fi
+    if [[ -z "${WEBHOOK_SECRET}" ]]; then
+        printf "WEBHOOK_SECRET environment variable needs to be set\n\n"
+        usage
+        exit 1
+    fi
+}
+
+operator_install() {
+    OPERATOR_INSTALL="${OPERATOR_INSTALL:-}"
+    if [[ "${OPERATOR_INSTALL,,}" == "true" ]]; then
+        printf "Installing operator\n"
+       kubectl --kubeconfig ${KUBECONFIG} apply -f https://raw.githubusercontent.com/openshift-pipelines/pipelines-as-code/stable/release.k8s.yaml
+    fi
+}
+
+mk_tmpdir () {
+  TMP_DIR="$(mktemp -d -t pac-pipelines-service.XXXXXXXXX)"
+  printf "Temporary directory created: ${TMP_DIR}\n"
+}
+
+kustomize () {
+
+    GIT_TOKEN=$(echo "${GIT_TOKEN}" | base64)
+    WEBHOOK_SECRET=$(echo "${WEBHOOK_SECRET}" | base64)
+
+    # Create a json patch for the repository
+    cat <<EOF > ${TMP_DIR}/patch-repo.yaml
+- op: replace
+  path: /spec/url
+  value: ${GITOPS_REPO}
+EOF
+
+    # Create a json patch for the secret
+    cat <<EOF > ${TMP_DIR}/patch-secret.yaml
+- op: replace
+  path: /data/provider.token
+  value: ${GIT_TOKEN}
+- op: replace
+  path: /data/webhook.secret
+  value: ${WEBHOOK_SECRET}
+EOF
+
+    if [[ "${OPERATOR_INSTALL,,}" == "true" ]]; then
```

Comment:
why the commas after OPERATOR_INSTALL?

Reply:
This is for making the comparison case insensitive. Apparently this is not working well with MacOS I will change it to:
 $(tr '[:upper:]' '[:lower:]' <<< "$OPERATOR_INSTALL")